packages/devtools_app/lib/src/charts/chart.dart

```diff
@@ -85,14 +86,20 @@ class ChartState extends State<Chart> with AutoDisposeMixin {
 
   @override
   Widget build(BuildContext context) {
-    return LayoutBuilder(
-      // Inner container
-      builder: (_, constraints) => Container(
-        width: constraints.widthConstraints().maxWidth,
-        height: constraints.widthConstraints().maxHeight,
+    // Need to wrap in a RepaintBoundary. Otherwise when entering in a button
```

Comment:
Nit: maybe just comment that the paint for the chart is somewhat expensive so a repaint boundary is worthwhile? Otherwise someone reading could be confused that has to do with the buttons rather than the general Flutter issue/feature that repaints will occur any time other content in the layer needs to repaint unless you add a repaint boundary.

Reply:
// Chart's Custom Painter (paint) can be expensive for lots of data points (10,000s).
    // A repaint boundary is necessary.
    // TODO(terry): Optimize the 10,000s of data points to just the number of pixels in the
    //              chart - this will make paint very fast.